include/revng/Yield/CallGraphs/Graph.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Support/BasicBlockID.h"
+#include "revng/Yield/Support/GraphLayout/Graphs.h"
+
+namespace yield::calls {
+
+struct Node {
+  BasicBlockID Address;
+  BasicBlockID NextAddress;
+
+  explicit Node(BasicBlockID Address = BasicBlockID::invalid()) :
+    Address(std::move(Address)) {}
+};
+
+struct Edge {
+  bool IsBackwards = false;
+};
+
+using PreLayoutGraph = layout::InputGraph<Node, Edge>;
+using PreLayoutNode = PreLayoutGraph::Node;
+
+using PostLayoutGraph = layout::OutputGraph<Node, Edge>;
+using PostLayoutNode = PostLayoutGraph::Node;
```

Comment:
Couldn't all these just be the output version and then we do the layout in-place? AFAICT the actual layout engine only works on internal graph, so it could update the place/route info in place.

Reply:
> the actual layout engine only works on internal graph

It only works on the output graph, but this trickery is more to allow users not having to deal with output graphs until the layouter is done, as per mattermost discussion. Closing this.